src/_guides/language/effective-dart/design.md

```diff
@@ -1219,103 +1204,90 @@ var names = people.map((person) => person.name);
 {:.bad-style}
 <?code-excerpt "misc/lib/effective_dart/design_bad.dart (func-expr-no-param-type)"?>
 {% prettify dart %}
-var names = people.map((Person person) {
-  return person.name;
-});
+var names = people.map((Person person) => person.name);
 {% endprettify %}
 
 In rare cases, the surrounding context is not precise enough to provide a type
 for one or more of the function's parameters. In those cases, you may need to
 annotate.
 
 
-### AVOID type arguments on generic invocations that can be inferred.
+### AVOID redundant type arguments on generic invocations.
 
-It's often useful to annotate *declarations* because they form the boundary of
-your API. But when instantiating an object or invoking a generic method, the
-type is often just noise. If inference produces the correct type, omit the
-annotation.
-
-When this rule is followed consistently, it helps draw the reader's attention to
-cases where inference does *not* produce the right type. The presence of the
-type argument sends a signal that that type is not otherwise pinned down by the
-surrounding code, and is telling you something useful.
+A type argument is redundant if inference would fill in the same type. If the
+invocation is the initializer for a type-annotated variable, or is an argument
+to a function, then inference usually fills in the type for you:
 
 {:.good-style}
-<?code-excerpt "misc/lib/effective_dart/design_good.dart (generic-invocation)"?>
+<?code-excerpt "misc/lib/effective_dart/design_good.dart (redundant-inferred)"?>
 {% prettify dart %}
-var strings = ["str", "ing"];
-var threes = new List.filled(3, 2);
-var pointThrees = threes.map((i) => i * 0.1);
+Set<String> things = new Set();
 {% endprettify %}
 
 {:.bad-style}
-<?code-excerpt "misc/lib/effective_dart/design_bad.dart (generic-invocation)"?>
+<?code-excerpt "misc/lib/effective_dart/design_bad.dart (redundant-inferred)"?>
 {% prettify dart %}
-var strings = <String>["str", "ing"];
-var threes = new List<int>.filled(3, 2);
-var pointThrees = threes.map<double>((i) => i * 0.1);
+Set<String> things = new Set<String>();
 {% endprettify %}
 
+Here, the type annotation on the variable is used to infer the type argument of
+constructor call in the initializer. In cases like this where inference produces
+the right type, it's usually abundantly obvious what the type argument will be
+and writing it again is just noise.
```

Comment:
Come to think of it, for sake of brievity, I'd consider dropping the sentence "In cases like this ... abundantly obvious ... just noise."

Reply:
Fair. Done. I thought it might help to provide some motivation, but it's probably overkill.